src/main/java/com/snowflake/kafka/connector/internal/streaming/DirectTopicPartitionChannel.java

```diff
@@ -689,19 +700,17 @@ private void resetChannelMetadataAfterRecovery(
       SnowflakeStreamingIngestChannel newChannel) {
     if (offsetRecoveredFromSnowflake == NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE) {
       LOGGER.info(
-          "{} Channel:{}, offset token is NULL",
+          "{} Channel:{}, offset token is NULL, will attempt to use offset managed by the connector"
+              + ", consumer offset: {}",
           streamingApiFallbackInvoker,
-          this.getChannelNameFormatV1());
+          this.getChannelNameFormatV1(),
+          this.latestConsumerOffset.get());
     }
 
-    final long offsetToResetInKafka = offsetRecoveredFromSnowflake + 1L;
-    if (offsetToResetInKafka == NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE) {
-      LOGGER.info(
-          "There is no offset registered for {} channel in Snowflake. Stop recovering the channel"
-              + " metadata.",
-          this.getChannelNameFormatV1());
-      return;
-    }
+    final long offsetToResetInKafka =
+        offsetRecoveredFromSnowflake == NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE
+            ? latestConsumerOffset.get()
+            : offsetRecoveredFromSnowflake + 1L;
```

Comment:
I expect we can fall into this method even when `latestConsumerOffset` has not been initialized yet. Even though there is a consumer group offset.
In these cases we don't want to reset consumer group offset to 0.
In the Buffered channel there is an additional check for this scenario:
```java
    final long offsetToResetInKafka =
        offsetRecoveredFromSnowflake == NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE
            ? latestConsumerOffset.get()
            : offsetRecoveredFromSnowflake + 1L;
    if (offsetToResetInKafka == NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE) {
      return;
    }
```